singlem/diamond_spkg_searcher.py

```diff
@@ -70,41 +72,57 @@ def _prefilter(self, diamond_database, read_files, is_reverse_reads, performance
             
             f = open(fasta_path, 'w+') # create tempfile in working directory
             f.close()
-            
-            cmd = "diamond blastx " \
-                  "--outfmt 6 qseqid full_qseq sseqid " \
-                  "--max-target-seqs 1 " \
-                  "--evalue 0.01 " \
-                  "%s " \
-                  "--threads %i " \
-                  "--query %s " \
-                  "--db %s " \
-                  "| tee >(sed 's/^/>/; s/\\t/\\n/; s/\\t.*//' > %s) " \
-                  "| awk '{print $1,$3}'" % (
-                      performance_parameters,
-                      self._num_threads,
-                      file,
-                      diamond_database,
-                      fasta_path)
-            logging.debug(cmd)
-
-            # Originially, we ran here via os.system rather than normal extern
-            # so reads can be piped in to singlem. However, this meant that
-            # errors and failed commands were ignored, sometimes causing
-            # successful return of singlem but an empty OTU table.
-            qseqid_sseqid = extern.run(cmd)
+
+            # DIAMOND command
+            # now with range culling, etc
+            cmd = [ 
+                "diamond", "blastx",
+                "--outfmt", "6", "qseqid", "full_qseq", "sseqid",
+                "--max-target-seqs", "0",
+                "--evalue", "0.01",
+                "--frameshift", "15",
+                "--range-culling",
+                "-k", "1",
```

Comment:
Earlier you have --max-target-seqs 0, but here you have -k 1 which probably explains why you're getting the same results from the former. --max-target-seqs and -k are actually the same parameter.